dashboard/README.md

```diff
@@ -1,2 +1,141 @@
-# Packit service web UI
+# Packit Service Web UI
 Development moved to separated [Dashboard project](https://github.com/packit-service/dashboard)
+
+
+## UI
+
+### Utils
+
+* SCSS is a superset of CSS with variable support and better query selection. It is compiled to _readable_ CSS.
+
+### UI/UX
+
+Patternfly has two official variants.
+
+* Patternfly HTML: Its a CSS Framework. Zero inbuilt support for JavaScript. User has to take care of interactive components + animations themselves.
+* Patternfly React: Complete UI toolkit. Components are interactive by default. Inbuilt support for charts and datatables, which follow a similar design pattern.
+
+Note: They also have a slack for community support.
+
+## Routing and Templating
+
+We have multiple options availible for routing (url handling, changing pages) and templating. 
+
+Patternfly has official React bindings so I'm just skipping discussion about React alternatives.
+
+
+### 1) Only React: 
+Since the dashboard will fetch all data from the API, we can make it entirely client side. Using Fetch/Axios, the browser will fetch the json directly from the API, no middleman involved. Controlling the routes in the URL is possible beacuse react-router uses the HTML5 History API. After compiling the JSX to JS, we will have static HTML/CSS/JS which can be served from apache, nginx etc or even GitHub pages.
+
+#### Pros
+* Serving static files is faster than wsgi + python. 
+* One language for everything.
+* Single page web app.
+* Using React + Patternfly's extra React only features like charts and datatables is leaner than importing Patternfly HTML + jQuery + some charts library + datatables library + other plugins.
+
+
+#### Cons
```

Comment:
do we need to tailor the API jsons for react?

Reply:
No, we'll just fetch the json and then parse it as a regular javascipt object in the dashboard